### PR TITLE
Biometric authentication

### PR DESCRIPTION
**Pull request by SaliienkoAleksandr**

### CODE REVIEW

File: Gekko/UI/AuthenticationViewController.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
+//
+//  AuthenticationViewController.swift
+//  Gekko
+//
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: Remove line until the author name

File: Gekko/UI/AuthenticationViewController.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+class AuthenticationViewController : UIViewController {
+    
+    // MARK: Overriden methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let userDefaults = UserDefaults.standard
+        
+        publicKey = userDefaults.string(forKey:BTCTradeUAAccountSettingsViewController.PublicKeySettingsKey)
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: Let's move these constant to UIUtils.swift

File: Gekko/UI/AuthenticationViewController.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        privateKey = userDefaults.string(forKey:BTCTradeUAAccountSettingsViewController.PrivateKeySettingsKey)
+        
+        self.setupMainIconImageView()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: Remove white space after ':' (and below also if needed)

File: Gekko/UI/AuthenticationViewController.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            make.centerX.equalTo(sizeOfMainView.width / 2)
+            make.centerY.equalTo(sizeOfMainView.height / 4)
+        }
+    }
+    
+    fileprivate var isAuthorized:Bool {
+        return publicKey != nil && !publicKey!.isEmpty && privateKey != nil && !privateKey!.isEmpty
+    }
+    
+    func authenticationWithTouchID() {
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: -> authenticate

File: Gekko/UI/AuthenticationViewController.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                self.navigateToMainViewController()
+            }
+            else {
+                self.enableFailLabel()
+            }
+        }
+    }
+    
+    fileprivate func navigateToMainViewController() {
+        DispatchQueue.main.async { [weak self] in
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: Seems that dispatch is not needed

File: Gekko/UI/AuthenticationViewController.swift [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            if self != nil {
+                self!.performSegue(withIdentifier: AuthenticationViewController.ShowNavigationControllerSegueName, sender: self)
+            }
+        }
+    }
+    
+    fileprivate func enableFailLabel() {
+        DispatchQueue.main.async { [weak self] in
```
Comment (SergiiMykhailov) [inline Gekko/UI/AuthenticationViewController.swift]: Move dispatch to place of invokation

File: Gekko/UI/BTCTradeUAAccountSettingsViewController.swift [hunk 7]
```diff
@@ -118,6 +118,12 @@ public static let PrivateKeySettingsKey = "Private Key"
 
     @IBOutlet weak var privateKeyLabel:UILabel?
     @IBOutlet weak var privateKeyField:UITextField?
+    
+    // MARK: Actions
+    
+    @IBAction func actionBackToMainView(_ sender: UISwipeGestureRecognizer) {
```
Comment (SergiiMykhailov) [inline Gekko/UI/BTCTradeUAAccountSettingsViewController.swift]: -> backButtonPressed

File: Gekko/UI/AuthenticationViewController.swift (hunk 5 above)
Comment (SaliienkoAleksandr) [reply Gekko/UI/AuthenticationViewController.swift]: No I think it has to be here because this method is called from background

**Pull request merged**